### ISSUE TITLE
Form Data/File Upload doesn't work

### ISSUE THREAD

**Issue by maahad767** [eadwinCode/django-ninja-extra#112]
The following code works perfectly fine. It works with the Form[...] wrap as well. 
```
@api.post("/upload", auth=None)
def upload(request, details: PasswordResetRequestIn, file: File[UploadedFile]):
    print(details.email)
    data = file.read()
    return {"name": file.name, "len": len(data)}
```
However, when the same code is written inside a controller, it doesn't work. I can't find any values in the parameters. The following code gives 422 error with profile_in missing, however, if it is wrapped with Form[...] then it doesn't give any error but it can't find any values passed to the parameter. 
```
@http_patch(response=ProfileOut)
    def update_profile(
        self, request, profile_in: ProfileIn, photo: File[UploadedFile] = None
    ):
...

```
I have tried using swagger, postman, curl, etc. none of them works if I write the code inside a controller. 
Let me know if more details are required.

**Comment by eadwinCode** [comment]
@maahad767 Can you please share the error log?

**Comment by maahad767** [comment]
Code(controller):
   ```
class PasswordResetRequestIn(Schema):
    email: str


 @http_patch("upload", auth=None, response=GenericResponse)
    def upload(
        self, request, details: PasswordResetRequestIn, file: File[UploadedFile]
    ):
        print(details.email)
        data = file.read()
        return {"name": file.name, "len": len(data)}
```
Response:
```
"PATCH - ProfileAPI[upload] /api/profile/upload" ([{'type': 'missing', 'loc': ('file', 'file'), 'msg': 'Field required'}, {'type': 'missing', 'loc': ('body', 'details'), 'msg': 'Field required'}],)
Unprocessable Entity: /api/profile/upload
```
Request:
<img width="1429" alt="image" src="https://github.com/eadwinCode/django-ninja-extra/assets/67506988/64e3e630-4555-41fb-a0c4-cb9b5dc1e6e9">
Response: 
<img width="1463" alt="image" src="https://github.com/eadwinCode/django-ninja-extra/assets/67506988/ca1071db-d9cc-4a09-bd21-77d5e30b19e6">

**Comment by eadwinCode** [comment]
@maahad767 I have looked into this here is my findings
1. using a `patch` http method for file uploads does not work. I tried this example [here](https://django-ninja.dev/guides/input/file-params/#uploading-files-with-extra-fields) using `PATCH` http method and it did work
2. If want to combine file with other schema, its good to annotate the schema with `Form`. That way django-ninja would spread the schema fields as an individual form field. For Example
```python
@http_post("/upload", auth=None)
def upload(request, details:Form[PasswordResetRequestIn], file: File[UploadedFile]):
    print(details.email)
    data = file.read()
    return {"name": file.name, "len": len(data)}
```
